nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSSession.java

```diff
@@ -905,17 +917,21 @@ protected void remove(String rootId) {
             }
         };
         collector.accept(rootState); // add the root node too
-        try (Stream<State> states = transaction.getDescendants(rootId, KEYS_RETENTION_ACTIVE_AND_PROXIES, 0)) {
+        try (Stream<State> states = transaction.getDescendants(rootId, KEYS_RETENTION_HOLD_AND_PROXIES, 0)) {
             states.forEach(collector);
         }
 
-        // if a subdocument is under active retention, removal fails
-        if (!retentionActiveIds.isEmpty()) {
-            if (retentionActiveIds.contains(rootId)) {
-                throw new DocumentExistsException("Cannot remove " + rootId + ", it is under active retention");
-            } else {
-                throw new DocumentExistsException("Cannot remove " + rootId + ", subdocument "
-                        + retentionActiveIds.iterator().next() + " is under active retention");
+        // if a subdocument is under retention / hold, removal fails
+        if (!undeletableIds.isEmpty()) {
+            // in tests we may want to delete everything
+            boolean allowDeleteUndeletable = Framework.isBooleanPropertyTrue("allowDeleteUndeletableDocuments");
```

Comment:
should we add `Framework.isTestModeSet()` to avoid any side effect of setting this property outer a test context

Reply:
It's also useful for cleanups in manual "functional" tests in a toy database without being in a "test mode set" setup, so I'll keep it.